lib/ember-test-helpers/-legacy-overrides.js

```diff
@@ -0,0 +1,90 @@
+import Ember from 'ember';
+
+import hasEmberVersion from './has-ember-version';
+
+export function preoutletRefactorSetupIntegrationForComponent() {
+  var module = this;
+  var context = this.context;
+
+  this.actionHooks = {};
+
+  context.dispatcher = this.container.lookup('event_dispatcher:main') || Ember.EventDispatcher.create();
+  context.dispatcher.setup({}, '#ember-testing');
+  context.actions = module.actionHooks;
+
+  (this.registry || this.container).register('component:-test-holder', Ember.Component.extend());
+
+  context.render = function(template) {
+    // in case `this.render` is called twice, make sure to teardown the first invocation
+    module.teardownComponent();
+
+    if (!template) {
+      throw new Error("in a component integration test you must pass a template to `render()`");
+    }
+    if (Ember.isArray(template)) {
+      template = template.join('');
+    }
+    if (typeof template === 'string') {
+      template = Ember.Handlebars.compile(template);
+    }
+    module.component = module.container.lookupFactory('component:-test-holder').create({
+      layout: template
+    });
+
+    module.component.set('context' ,context);
+    module.component.set('controller', context);
+
+    Ember.run(function() {
+      module.component.appendTo('#ember-testing');
+    });
+  };
+
+  context.$ = function() {
+    return module.component.$.apply(module.component, arguments);
+  };
+
+  context.set = function(key, value) {
+    var ret = Ember.run(function() {
+      return Ember.set(context, key, value);
+    });
+
+    if (hasEmberVersion(2,0)) {
+      return ret;
+    }
+  };
+
+  context.setProperties = function(hash) {
+    var ret = Ember.run(function() {
+      return Ember.setProperties(context, hash);
```

Comment:
Only in Ember 2.0+, it is intended to emulate what `Ember.setProperties` does: https://github.com/emberjs/ember.js/blob/dc4b60be5849a6dad2cc464822294a435e08e31c/packages/ember-metal/lib/set_properties.js#L37